browser/plugins/network-breadcrumbs.js

```diff
@@ -0,0 +1,144 @@
+/* global XMLHttpRequest */
+const breadcrumbType = 'network'
+let restoreFunctions = []
+let client
+
+/*
+ * Leaves breadcrumbs when network requests occur
+ */
+module.exports = {
+  init: (_client) => {
+    client = _client
+    monkeyPatchXMLHttpRequest()
+    monkeyPatchFetch()
+  },
+  destroy: () => {
+    restoreFunctions.forEach(restore => restore())
+    restoreFunctions = []
+  },
+  configSchema: {
+    networkBreadcrumbsEnabled: {
+      defaultValue: () => undefined,
+      validate: (value) => value === true || value === false || value === undefined,
+      message: '(boolean) networkBreadcrumbsEnabled should be true or false'
+    }
+  }
+}
+
+// ------------------------------------------------------------------------------------------------
+// XMLHttpRequest and XDomainRequest monkey patch
+// ------------------------------------------------------------------------------------------------
+// create keys to safely store metadata on the request object
+const requestSetupKey = 'BUGSNAG:SETUP'
+const requestUrlKey = 'BUGSNAG:REQUEST_URL'
+const requestMethodKey = 'BUGSNAG:REQUEST_METHOD'
+
+// generic function for monkey patching both XMLHttpRequest and XDomainRequest
+function monkeyPatchXMLHttpRequest () {
+  if (!('addEventListener' in XMLHttpRequest.prototype)) return
+  let nativeOpen = XMLHttpRequest.prototype.open
+
+  // override native open()
+  XMLHttpRequest.prototype.open = function open (method, url) {
+    // store url and HTTP method for later
+    this[requestUrlKey] = url
+    this[requestMethodKey] = method
+
+    // if we have already setup listeners, it means open() was called twice, we need to remove
+    // the listeners and recreate them
+    if (this[requestSetupKey]) {
+      this.removeEventListener('load', handleXHRLoad)
+      this.removeEventListener('error', handleXHRError)
+    }
+
+    // attach load event listener
+    this.addEventListener('load', handleXHRLoad)
+    // attach error event listener
+    this.addEventListener('error', handleXHRError)
+
+    this[requestSetupKey] = true
+
+    nativeOpen.apply(this, arguments)
+  }
+
+  restoreFunctions.push(() => {
+    XMLHttpRequest.prototype.open = nativeOpen
+  })
+}
+
+function handleXHRLoad () {
+  if (this[requestUrlKey] === client.notifier.url) {
+    // don't leave a network breadcrumb from bugsnag notify calls
+    return
+  }
+  let metaData = {
+    status: this.status,
+    request: `${this[requestMethodKey]} ${this[requestUrlKey]}`
+  }
+  if (this.status >= 400) {
+    // contacted server but got an error response
+    client.leaveBreadcrumb('XMLHttpRequest failed', metaData, breadcrumbType)
+  } else {
+    client.leaveBreadcrumb('XMLHttpRequest succeeded', metaData, breadcrumbType)
+  }
+}
+
+function handleXHRError () {
+  if (this[requestUrlKey] === client.notifier.url) {
```

Comment:
I think you want `client.config.endpoint`

`client.notifier.url='https://github.com/bugsnag/bugsnag-js'` is the github url for the notifier

Additionally you'll want `client.config.sessionEndpoint` if session tracking is enabled.